resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/kubernetes/tpr/sparkJobResourceController.scala

```diff
@@ -0,0 +1,316 @@
+    val payload = parse(write(resourceObject))
+    val requestBody = RequestBody
+      .create(MediaType.parse("application/json"), compact(render(payload)))
+    val requestPathSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs")
+    val url = generateHttpUrl(requestPathSegments)
+    val request = new Request.Builder()
+      .post(requestBody)
+      .url(url)
+      .build()
+
+    logDebug(s"Create SparkJobResource Request: $request")
+    var response: Response = null
+    try {
+      response = httpClient.newCall(request).execute()
+    } catch {
+      case x: IOException =>
+        val msg =
+          s"Failed to post resource $name. ${x.getMessage}. ${compact(render(payload))}"
+        logError(msg)
+        response.close()
+        throw new SparkException(msg)
+    }
+    completeRequestWithExceptionIfNotSuccessful(
+      "post",
+      response,
+      Option(Seq(name, response.toString, compact(render(payload)))))
+    logDebug(s"Successfully posted resource $name: " +
+      s"${pretty(render(parse(write(resourceObject))))}")
+    response.body().close()
+  }
+
+  /**
+   * This method deletes a resource of kind "SparkJob" with the specified name
+   * @return Unit
+   * @throws IOException Due to failure from execute call
+   * @throws SparkException when DELETE request is unsuccessful
+   */
+  override def deleteJobObject(tprObjectName: String): Unit = {
+    val requestPathSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", tprObjectName)
+    val url = generateHttpUrl(requestPathSegments)
+    val request = new Request.Builder()
+      .delete()
+      .url(url)
+      .build()
+
+    logDebug(s"Delete Request: $request")
+    var response: Response = null
+    try {
+      response = httpClient.newCall(request).execute()
+    } catch {
+      case x: IOException =>
+        val msg =
+          s"Failed to delete resource. ${x.getMessage}."
+        logError(msg)
+        response.close()
+        throw new SparkException(msg)
+    }
+    completeRequestWithExceptionIfNotSuccessful(
+      "delete",
+      response,
+      Option(Seq(tprObjectName, response.message(), request.toString)))
+
+    response.body().close()
+    logInfo(s"Successfully deleted resource $tprObjectName")
+  }
+
+  /**
+   * This method GETS a resource of kind "SparkJob" with the given name
+   * @return SparkJobState
+   * @throws IOException Due to failure from execute call
+   * @throws SparkException when GET request is unsuccessful
+   */
+  override def getJobObject(name: String): SparkJobState = {
+    val requestPathSegments = Seq(
+      "apis", TPR_API_GROUP, TPR_API_VERSION, "namespaces", namespace, "sparkjobs", name)
+    val url = generateHttpUrl(requestPathSegments)
+    val request = new Request.Builder()
+      .get()
+      .url(url)
+      .build()
+
+    logDebug(s"Get Request: $request")
+    var response: Response = null
```

Comment:
You can use

`val response = try {
    httpCLient.newCall(request).execute()
  }`